src/preact/vote-buttons.js

```diff
@@ -21,15 +21,8 @@ import {h, Component} from 'preact';
 
 export default class VoteButtons extends Component {
 
-  constructor (props) {
-    super(props);
-
-    this.state = {
-      score: this.props.score
-    };
-
-    this.onVoteUp = this.onVoteUp.bind(this);
-    this.onVoteDown = this.onVoteDown.bind(this);
+  componentWillMount () {
+    this.state.score = this.props.score;
   }
 
```

Comment:
I’ve since read more about this topic and based on what I’ve learned, this `VoteButtons` component should not hold any state at all, as it’s a presentational component, meaning it receives both data and callbacks via `props` from its parent container component.

Good to hear that `.bind()` is cheap! I’ve tested the `autobind-decorator` package (it’s only a few KBs, uncompressed) and it gets the job done, so I’d recommend using that over `.bind()` or `::`.